core/domain/fs_domain.py

```diff
@@ -330,13 +464,13 @@ def get(self, filepath):
             raise IOError('File %s not found.' % (filepath))
         return file_stream.read()
 
-    def commit(self, filepath, raw_bytes, mimetype=None):
+    def commit(self, filepath: str, raw_bytes: bytes, mimetype: str='') -> None:
```

Comment:
```suggestion
Again why is `mimetype` set to `''`
```

Reply:
Then how am I supposed to define an empty string as the default value here? I can't use optional(str) as it pops the eror that the function returning str is expecting optional(str). I can't use NULL as string can take the type NULL. So I am not sure how will I define an empty value with string as the type.

Reply:
Sorry I forgot again what are the errors here. I think you will need to think about this, `''` is not acceptable, I think you might use `None` if you refactor this and all the callsites.